src/api/spec/controllers/webui/user_controller_spec.rb

```diff
@@ -217,12 +222,22 @@
     context "when home project creation disabled" do
       before do
         allow(Configuration).to receive(:allow_user_to_create_home_project).and_return(false)
-        post :register, params: { login: new_user.login, email: new_user.email, password: 'buildservice' }
+        post :register, params: { login: new_user.login, email: new_user.email, password: 'buildservice', password_repeat: 'buildservice' }
       end
 
       it { expect(flash[:success]).to eq("The account '#{new_user.login}' is now active.") }
       it { expect(response).to redirect_to root_path }
     end
+
+    context "when passwords do not match" do
+      before do
+        allow(Configuration).to receive(:allow_user_to_create_home_project).and_return(false)
+        post :register, params: { login: new_user.login, email: new_user.email, password: 'buildservice', password_repeat: 'abc' }
+      end
+
+      it { expect(flash[:error]).to eq("The passwords do not match.") }
+      it { expect(response).to redirect_to root_path }
```

Comment:
I would tests this similarly of how it is done here:

https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/user_controller_spec.rb#L78

because if nor you are using the `fallback_location` but not really the redirect back.